### PR TITLE
infra: make it possible to cache docker containers build

### PR DESCRIPTION
**Pull request by DavidKorczynski**

### CODE REVIEW

File: infra/helper.py [hunk 1]
```diff
@@ -721,9 +721,14 @@ def docker_run(run_args, print_output=True, architecture='x86_64'):
...
   command = [
-      'docker', 'run', '--rm', '--privileged', '--shm-size=2g', '--platform',
+      'docker', 'run', '--privileged', '--shm-size=2g', '--platform',
       platform
   ]
+  if not os.getenv('OSS_FUZZ_SAVE_CONTAINERS_NAME'):
+      command.append('--rm')
+  else:
+      command.append('--name')
+      command.append(os.getenv('OSS_FUZZ_SAVE_CONTAINERS_NAME'))
```
Comment (DonggeLiu) [inline infra/helper.py:731]: nit: flip the logic for simplicity.
```python
if os.getenv('OSS_FUZZ_SAVE_CONTAINERS_NAME'):
  ...
else:
  ...
```

**Review by DonggeLiu — APPROVED** [review]

File: infra/helper.py (hunk 1 above)
Comment (DavidKorczynski) [reply infra/helper.py]: Sounds good, done

Comment (DonggeLiu) [reply infra/helper.py]: Thanks, let's wait for @oliverchang to have a final look.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**